test/common_plat/validation/api/pktio/pktio.c

```diff
@@ -1088,6 +1091,27 @@ void pktio_test_mac(void)
 	mac_len = odp_pktio_mac_addr(pktio, mac_addr, 2);
 	CU_ASSERT(mac_len < 0);
 
+	CU_ASSERT_FATAL(odp_pktio_capability(pktio, &capa) == 0);
+	if (!capa.set_op.op.mac_addr) {
+		printf("\n mac address set not supported for %s!\n",
+		       iface_name[0]);
+		ret = odp_pktio_close(pktio);
+		CU_ASSERT(ret == 0);
```

Comment:
I'd organize this the other way - move MAC addr set test code inside the if-clause, so that the same pktio_close() path would be used at the end of the test.